### ISSUE TITLE
seastar_memory - oversized allocation: 1564672 bytes in alternator, while preparing a response

### ISSUE THREAD

**Issue by Felix-zhoux** [scylladb/scylla#8661]
*Installation details*
Scylla version (or git commit hash): 4.4.1

I found the following error in the scylla log:
```
May 16 17:36:26 node02 scylla[346836]:  [shard 51] seastar_memory - oversized allocation: 1564672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at    0x3c7cbbe
                                                                        0x3c7d030
                                                                        0x3c7d3b8
                                                                        0x38ac21b
                                                                        0x38ada23
                                                                        0x38b0a60
                                                                        0x123dc96
                                                                        0x12315db
                                                                        0x12315db
                                                                        0x123141a
                                                                        0x12315db
                                                                        0x1230ebb
                                                                        0x33f03fc
                                                                        0x3ba98e8
                                                                        0x34919fc
                                                                        0x3494dd0
                                                                        0x38f307f
                                                                        0x38f4267
                                                                        0x3912928
                                                                        0x38be9ca
                                                                        /opt/scylladb/libreloc/libpthread.so.0+0x93f8
                                                                        /opt/scylladb/libreloc/libc.so.6+0x101902
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35, seastar::future<std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >::then_impl_nrvo<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > >(alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&&)::{lambda(seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >&&, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, seastar::future_state<std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >&&)#1}, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >::finally_body<alternator::server::handle_api_request(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >&&)::$_1::operator()()::{lambda()#1}::operator()()::{lambda(std::unique_ptr<service::client_state, std::default_delete<service> >&)#1}::operator()(std::unique_ptr<service::client_state, std::default_delete<service> >)::{lambda(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>)#1}::operator()(std::chrono::_V2)::{lambda(rapidjson::GenericValue<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::UTF8<char>, seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::CrtAllocator>)#1}::operator()(seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::UTF8)::{lambda()#1}, false>, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >::then_wrapped_nrvo<seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >, rapidjson::GenericValue<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::UTF8<char>, seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::CrtAllocator> >(rapidjson::GenericValue<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::UTF8<char>, seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::CrtAllocator>&&)::{lambda(seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >&&, rapidjson::GenericValue<seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::UTF8<char>, seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock>::CrtAllocator>&, seastar::future_state<std::variant<seastar::json::json_return_type, alternator::api_error> >&&)#1}, std::variant<seastar::json::json_return_type, alternator::api_error> >
                                                                        --------
                                                                        seastar::internal::do_with_state<std::tuple<std::unique_ptr<service::client_state, std::default_delete<service::client_state> > >, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > >
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >::finally_body<seastar::internal::invoke_func_with_gate<alternator::server::handle_api_request(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >&&)::$_1::operator()()::{lambda()#1}>(seastar::gate&, alternator::server::handle_api_request(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >&&)::$_1::operator()()::{lambda()#1}&&)::{lambda()#1}, false>, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >::then_wrapped_nrvo<seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >, {lambda()#1}>({lambda()#1}&&)::{lambda(seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >&&, {lambda()#1}&, seastar::future_state<std::variant<seastar::json::json_return_type, alternator::api_error> >&&)#1}, std::variant<seastar::json::json_return_type, alternator::api_error> >
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >, alternator::api_handler::api_handler(std::function<seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > (std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)> const&)::{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}::operator()(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >) const::{lambda(seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >)#1}, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >::then_wrapped_nrvo<seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >, {lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}>({lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}&&)::{lambda(seastar::internal::promise_base_with_type<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >&&, {lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}&, seastar::future_state<std::variant<seastar::json::json_return_type, alternator::api_error> >&&)#1}, std::variant<seastar::json::json_return_type, alternator::api_error> >
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >, alternator::api_handler::handle(seastar::basic_sstring<char, unsigned int, 15u, true> const&, std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)::{lambda(std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}, seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >::then_impl_nrvo<{lambda(std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}, seastar::future>({lambda(std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}&&)::{lambda(seastar::internal::promise_base_with_type<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >&&, {lambda(std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> >)#1}&, seastar::future_state<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >&&)#1}, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >, seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >::handle_exception<std::function<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > (std::__exception_ptr::exception_ptr)>&>(std::function<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > (std::__exception_ptr::exception_ptr)>&)::{lambda(auto:1&&)#1}, seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >::then_wrapped_nrvo<seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >, {lambda(auto:1&&)#1}>({lambda(auto:1&&)#1}&&)::{lambda(seastar::internal::promise_base_with_type<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >&&, {lambda(auto:1&&)#1}&, seastar::future_state<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >&&)#1}, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<bool>, seastar::httpd::connection::generate_reply(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)::$_19, seastar::future<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >::then_impl_nrvo<seastar::httpd::connection::generate_reply(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)::$_19, seastar::future<bool> >(seastar::httpd::connection::generate_reply(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)::$_19&&)::{lambda(seastar::internal::promise_base_with_type<bool>&&, seastar::httpd::connection::generate_reply(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)::$_19&, seastar::future_state<std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >&&)#1}, std::unique_ptr<seastar::httpd::reply, std::default_delete<seastar::httpd::reply> > >
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::httpd::connection::read_one()::$_12::operator()()::{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)#1}::operator()(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >) const::{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)#1}::operator()(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >) const::{lambda(bool)#1}, seastar::future<bool>::then_impl_nrvo<{lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)#1}, {lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)#1}<void> >({lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)#1}&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, {lambda(std::unique_ptr<seastar::httpd::request, std::default_delete<seastar::httpd::request> >)#1}&, seastar::future_state<bool>&&)#1}, bool>
                                                                        --------
                                                                        seastar::internal::do_until_state<seastar::httpd::connection::read()::$_29, seastar::httpd::connection::read()::$_28>
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::httpd::connection::read()::$_9, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::httpd::connection::read()::$_9>(seastar::httpd::connection::read()::$_9&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::httpd::connection::read()::$_9&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
                                                                        --------
                                                                        seastar::continuation<seastar::internal::promise_base_with_type<void>, seastar::future<void>::finally_body<seastar::httpd::connection::read()::$_10, true>, seastar::future<void>::then_wrapped_nrvo<seastar::future<void>, seastar::future<void>::finally_body<seastar::httpd::connection::read()::$_10, true> >(seastar::future<void>::finally_body<seastar::httpd::connection::read()::$_10, true>&&)::{lambda(seastar::internal::promise_base_with_type<void>&&, seastar::future<void>::finally_body<seastar::httpd::connection::read()::$_10, true>&, seastar::future_state<seastar::internal::monostate>&&)#1}, void>
```

decoded log
```
/usr/lib/debug/opt/scylladb/libexec/scylla-4.4.1-0.20210413.00da6b5e9.x86_64.debug: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for GNU/Linux 3.2.0, BuildID[sha1]=f1b7ffd311c0488984c05a1b3c0d658551419367, not stripped

May 16 17:36:26 node02 scylla[346836]:  [shard 51] seastar_memory - oversized allocation: 1564672 bytes. This is non-fatal, but could lead to latency and/or fragmentation issues. Please report: at   
[Backtrace #0]
void seastar::backtrace<seastar::current_backtrace_tasklocal()::$_3>(seastar::current_backtrace_tasklocal()::$_3&&) at ./build/release/seastar/./seastar/include/seastar/util/backtrace.hh:59
 (inlined by) seastar::current_backtrace_tasklocal() at ./build/release/seastar/./seastar/src/util/backtrace.cc:86
seastar::current_tasktrace() at ./build/release/seastar/./seastar/src/util/backtrace.cc:135
seastar::current_backtrace() at ./build/release/seastar/./seastar/src/util/backtrace.cc:168
seastar::memory::cpu_pages::warn_large_allocation(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:752
seastar::memory::cpu_pages::check_large_allocation(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:760
 (inlined by) seastar::memory::cpu_pages::allocate_large(unsigned int) at ./build/release/seastar/./seastar/src/core/memory.cc:766
 (inlined by) seastar::memory::allocate_large(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1318
 (inlined by) seastar::memory::allocate(unsigned long) at ./build/release/seastar/./seastar/src/core/memory.cc:1380
malloc at ./build/release/seastar/./seastar/src/core/memory.cc:1806
 (inlined by) realloc at ./build/release/seastar/./seastar/src/core/memory.cc:1893
rapidjson::CrtAllocator::Realloc(void*, unsigned long, unsigned long) at /usr/include/rapidjson/internal/../allocators.h:77
 (inlined by) rapidjson::internal::Stack<rapidjson::CrtAllocator>::Resize(unsigned long) at /usr/include/rapidjson/internal/stack.h:201
 (inlined by) void rapidjson::internal::Stack<rapidjson::CrtAllocator>::Expand<char>(unsigned long) at /usr/include/rapidjson/internal/stack.h:196
 (inlined by) void rapidjson::internal::Stack<rapidjson::CrtAllocator>::Reserve<char>(unsigned long) at /usr/include/rapidjson/internal/stack.h:118
 (inlined by) rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>::Reserve(unsigned long) at /usr/include/rapidjson/stringbuffer.h:68
 (inlined by) void rapidjson::PutReserve<rapidjson::UTF8<char>, rapidjson::CrtAllocator>(rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>&, unsigned long) at /usr/include/rapidjson/stringbuffer.h:97
 (inlined by) rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>::WriteString(char const*, unsigned int) at /usr/include/rapidjson/writer.h:360
bool rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator>::Accept<rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false> >(rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false>&) const at /usr/include/rapidjson/document.h:1781
bool rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator>::Accept<rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false> >(rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false>&) const at /usr/include/rapidjson/document.h:1781
bool rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator>::Accept<rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false> >(rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false>&) const at /usr/include/rapidjson/document.h:1790
bool rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator>::Accept<rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false> >(rjson::guarded_yieldable_json_handler<rapidjson::Writer<rapidjson::GenericStringBuffer<rapidjson::UTF8<char>, rapidjson::CrtAllocator>, rapidjson::UTF8<char>, rapidjson::UTF8<char>, rapidjson::CrtAllocator, 0u>, false>&) const at /usr/include/rapidjson/document.h:1781
rjson::print[abi:cxx11](rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const&) at ./utils/rjson.cc:126
alternator::make_jsonable::to_json[abi:cxx11]() const at ./alternator/executor.cc:80
seastar::json::formatter::to_json(seastar::json::jsonable const&) at ./build/release/seastar/./seastar/src/json/formatter.cc:109
json_return_type<alternator::make_jsonable> at ././seastar/include/seastar/json/json_elements.hh:294
 (inlined by) _Uninitialized<alternator::make_jsonable> at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:242
 (inlined by) _Variadic_union<alternator::make_jsonable> at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:336
 (inlined by) _Variant_storage<0, alternator::make_jsonable> at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:396
 (inlined by) _Variant_storage at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:506
 (inlined by) _Variant_storage at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:534
 (inlined by) _Variant_storage at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:594
 (inlined by) _Variant_storage at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:652
 (inlined by) _Variant_base<0, alternator::make_jsonable> at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:704
 (inlined by) variant<0, alternator::make_jsonable, seastar::json::json_return_type, void> at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:1400
 (inlined by) variant<alternator::make_jsonable, void, void, seastar::json::json_return_type, void> at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/variant:1371
 (inlined by) _ZN7seastar8internal26uninitialized_wrapper_baseISt7variantIJNS_4json16json_return_typeEN10alternator9api_errorEEELb0EE17uninitialized_setIJNS5_13make_jsonableEEEENSt9enable_ifIXntsr3stdE9is_same_vISt5tupleIJDpNSt9remove_cvIT_E4typeEEESC_IJSC_IJS7_EEEEEEvE4typeEDpOSE_ at ././seastar/include/seastar/core/future.hh:308
 (inlined by) future_state<alternator::make_jsonable> at ././seastar/include/seastar/core/future.hh:627
 (inlined by) future<alternator::make_jsonable> at ././seastar/include/seastar/core/future.hh:1354
 (inlined by) seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > seastar::make_ready_future<std::variant<seastar::json::json_return_type, alternator::api_error>, alternator::make_jsonable>(alternator::make_jsonable&&) at ././seastar/include/seastar/core/future.hh:2049
 (inlined by) operator() at ./alternator/executor.cc:3241
seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > std::__invoke_impl<seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> >, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >(std::__invoke_other, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> >&&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/invoke.h:60
 (inlined by) std::__invoke_result<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >::type std::__invoke<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >(alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> >&&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/invoke.h:95
 (inlined by) std::invoke_result<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >::type std::invoke<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >(alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> >&&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/functional:88
 (inlined by) auto seastar::internal::future_invoke<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >(alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> >&&) at ././seastar/include/seastar/core/future.hh:1211
 (inlined by) operator() at ././seastar/include/seastar/core/future.hh:1582
 (inlined by) void seastar::futurize<seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > >::satisfy_with_result_of<seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > seastar::future<std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >::then_impl_nrvo<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > >(alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&&)::{lambda(seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >&&, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, seastar::future_state<std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >&&)#1}::operator()(seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >&&, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, seastar::future_state<std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >&&) const::{lambda()#1}>(seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >&&, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&&) at ././seastar/include/seastar/core/future.hh:2120
 (inlined by) operator() at ././seastar/include/seastar/core/future.hh:1575
 (inlined by) seastar::continuation<seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > seastar::future<std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >::then_impl_nrvo<alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35, seastar::future<std::variant<seastar::json::json_return_type, alternator::api_error> > >(alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&&)::{lambda(seastar::internal::promise_base_with_type<std::variant<seastar::json::json_return_type, alternator::api_error> >&&, alternator::do_query(service::storage_proxy&, seastar::lw_shared_ptr<schema const>, rapidjson::GenericValue<rapidjson::UTF8<char>, rapidjson::CrtAllocator> const*, std::vector<nonwrapping_interval<dht::ring_position>, std::allocator<nonwrapping_interval<dht::ring_position> > >&&, std::vector<nonwrapping_interval<clustering_key_prefix>, std::allocator<nonwrapping_interval<clustering_key_prefix> > >&&, std::unordered_map<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, alternator::attribute_path_map_node<std::monostate>, std::hash<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::equal_to<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const, alternator::attribute_path_map_node<std::monostate> > > >&&, unsigned int, db::consistency_level, alternator::filter&&, enum_set<super_enum<query::partition_slice::option, (query::partition_slice::option)0, (query::partition_slice::option)1, (query::partition_slice::option)2, (query::partition_slice::option)3, (query::partition_slice::option)4, (query::partition_slice::option)5, (query::partition_slice::option)6, (query::partition_slice::option)7, (query::partition_slice::option)8, (query::partition_slice::option)9, (query::partition_slice::option)10, (query::partition_slice::option)11> >, service::client_state&, cql3::cql_stats&, tracing::trace_state_ptr, service_permit)::$_35&, seastar::future_state<std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >&&)#1}, std::unique_ptr<cql3::result_set, std::default_delete<cql3::result_set> > >::run_and_dispose() at ././seastar/include/seastar/core/future.hh:767
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at ./build/release/seastar/./seastar/src/core/reactor.cc:2220
 (inlined by) seastar::reactor::run_some_tasks() at ./build/release/seastar/./seastar/src/core/reactor.cc:2629
seastar::reactor::run() at ./build/release/seastar/./seastar/src/core/reactor.cc:2788
operator() at ./build/release/seastar/./seastar/src/core/reactor.cc:3979
 (inlined by) void std::__invoke_impl<void, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&>(std::__invoke_other, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/invoke.h:60
 (inlined by) std::enable_if<is_invocable_r_v<void, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&>, void>::type std::__invoke_r<void, seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&>(seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/invoke.h:110
 (inlined by) std::_Function_handler<void (), seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::$_97>::_M_invoke(std::_Any_data const&) at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/std_function.h:291
std::function<void ()>::operator()() const at /usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/std_function.h:622
 (inlined by) seastar::posix_thread::start_routine(void*) at ./build/release/seastar/./seastar/src/core/posix.cc:60
/opt/scylladb/libreloc/libpthread.so.0: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), BuildID[sha1]=e745060dff7db1a75759f6a50a20b14f69eeb050, for GNU/Linux 3.2.0, stripped

__libpthread_freeres at ??:?
/opt/scylladb/libreloc/libc.so.6: ELF 64-bit LSB shared object, x86-64, version 1 (GNU/Linux), dynamically linked (uses shared libs), BuildID[sha1]=310641736c5d52be58f3e9efa3590348b50503c5, for GNU/Linux 3.2.0, stripped

clone at ??:?

```

**Comment by avikivity** [comment]
```c++

std::string print(const rjson::value& value, size_t max_nested_level) {
    string_buffer buffer;
    guarded_yieldable_json_handler<writer, false> writer(buffer, max_nested_level);
    value.Accept(writer);
    return std::string(buffer.GetString());
}
```

buffer grows returning a large result. Maybe we can replace string_buffer with a chunked class (and return a chunked thing rather than std::string).

**Comment by nyh** [comment]
This is a duplicate of #8522 - ("Alternator should stream JSON response object - not convert it to a contiguous string"). 

The problem is that Alternator takes a JSON response object (which isn't contiguous) and to return it as an HTTP response, it serializes it to one big contiguous string (which is what the HTTP server expects), and returns that to the HTTP server. 

In issue #8522 I suggested that we don't do that serialization step at all - we should use the HTTP server's alternator interface where the application can write to a stream instead of producing one contiguous stream. However, doing this is difficult - it means futurizing rapidjson's JSON printer, so it can wait to write into the stream. @avikivity  suggested here a nice compromise that we should indeed consider:

As a first step we can convert the JSON printing to write into a non-contiguous string (unfuturized), and then write this non-contiguous string to the output stream. This will solve the contiguous allocation problem, but will not solve 1. the unpreemptable print() code, and 2. the waste of memory to hold two copies of the JSON in memory at the same time.

I'll copy this suggestion to #8522 and consider implementing this workaround before we can have the perfect solution.
